crates/spfs/src/env.rs

```diff
@@ -5,16 +5,51 @@
 //! Functions related to the setup and teardown of the spfs runtime environment
 //! and related system namespacing
 
+use std::ffi::{CStr, CString};
 use std::path::{Path, PathBuf};
 
+use linux_raw_sys::general::fsconfig_command;
+use linux_syscall::{
+    syscall,
+    SYS_fsconfig,
+    SYS_fsmount,
+    SYS_fsopen,
+    SYS_mount_setattr,
+    SYS_move_mount,
+};
+
 use super::runtime;
 use crate::{which, Error, Result};
 
 pub const SPFS_DIR: &str = "/spfs";
 pub const SPFS_DIR_PREFIX: &str = "/spfs/";
+const SPFS_DIR_CSTR: &CStr = c"/spfs";
 
 const NONE: Option<&str> = None;
 
+// Linux syscall constants from /usr/include/linux/mount.h.
+const FSCONFIG_SET_STRING: u32 = fsconfig_command::FSCONFIG_SET_STRING as u32;
+const FSCONFIG_SET_FLAG: u32 = fsconfig_command::FSCONFIG_SET_FLAG as u32;
+const FSCONFIG_CMD_CREATE: u32 = fsconfig_command::FSCONFIG_CMD_CREATE as u32;
+const FSOPEN_CLOEXEC: u32 = 0x00000001;
+const FSMOUNT_CLOEXEC: u32 = 0x00000001;
+const MOUNT_ATTR_RDONLY: u32 = 0x00000001;
+const MOUNT_ATTR_SIZE_VER0: u32 = 32;
+const MOVE_MOUNT_F_EMPTY_PATH: u32 = 0x00000004;
+
+// Linux fcntl constants from /usr/include/fcntl.h.
+const AT_EMPTY_PATH: u32 = 0x1000;
+const AT_FDCWD: i32 = -100;
+
+/// This struct matches struct mount_attr from /usr/include/linux/mount.h.
+#[repr(C)]
+struct MountAttr {
+    pub attr_set: u64,
+    pub attr_clr: u64,
+    pub propagation: u64,
+    pub userns_fd: u64,
+}
```

Comment:
I was using an older version of linux-raw-sys when I first wrote this. It looks like the latest version (0.8.0) has support for this struct:
https://docs.rs/linux-raw-sys/0.8.0/linux_raw_sys/general/struct.mount_attr.html

Ignore this while reviewing. I can take a stab at replacing this struct with the one from linux-raw-sys soon.